App/Services/MutagenController.cs

```diff
@@ -3,36 +3,83 @@
 using System.ComponentModel.DataAnnotations;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Coder.Desktop.App.Models;
 using Coder.Desktop.MutagenSdk;
 using Coder.Desktop.MutagenSdk.Proto.Selection;
 using Coder.Desktop.MutagenSdk.Proto.Service.Daemon;
+using Coder.Desktop.MutagenSdk.Proto.Service.Prompting;
 using Coder.Desktop.MutagenSdk.Proto.Service.Synchronization;
+using Coder.Desktop.MutagenSdk.Proto.Synchronization;
+using Coder.Desktop.MutagenSdk.Proto.Url;
 using Coder.Desktop.Vpn.Utilities;
+using Grpc.Core;
 using Microsoft.Extensions.Options;
-using TerminateRequest = Coder.Desktop.MutagenSdk.Proto.Service.Daemon.TerminateRequest;
+using DaemonTerminateRequest = Coder.Desktop.MutagenSdk.Proto.Service.Daemon.TerminateRequest;
+using MutagenProtocol = Coder.Desktop.MutagenSdk.Proto.Url.Protocol;
+using SynchronizationTerminateRequest = Coder.Desktop.MutagenSdk.Proto.Service.Synchronization.TerminateRequest;
 
 namespace Coder.Desktop.App.Services;
 
-public class CreateSyncSessionRequest
+public enum CreateSyncSessionRequestEndpointProtocol
+{
+    Local,
+    Ssh,
+}
+
+public class CreateSyncSessionRequestEndpoint
```

Comment:
Just wondering out loud: these names are getting really long... is it C# style to namespace them rather than have a long prefix?

Reply:
I'm not sure what the convention is. Another option would be to move these inside the "parent type" e.g. `new CreateSyncSessionRequest.Endpoint()`.

Reply:
That makes sense to me since they are only used to fully define the `CreateSyncSessionRequest` type

Reply:
Renamed these to:
- `CreateSyncSessionRequest.Endpoint`
- `CreateSyncSessionRequest.Endpoint.ProtocolKind` (can't be `Protocol` or it conflicts with the property with the same name)

So the names are longer when using them, but they're segregated and not cluttering up the main namespace